crypto/aes.go

```diff
@@ -0,0 +1,155 @@
+package crypto
+
+import (
+	"bufio"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/d1str0/pkcs7"
+	"github.com/golang/glog"
+)
+
+func LoadPrivateKey(privateKeyBase64 string) (*rsa.PrivateKey, error) {
+	privateKey, err := base64.StdEncoding.DecodeString(privateKeyBase64)
+	if err != nil {
+		return nil, fmt.Errorf("file-decrypt: error decoding private key: %v", err)
+	}
+
+	block, _ := pem.Decode(privateKey)
+	if block == nil {
+		return nil, fmt.Errorf("file-decrypt: error decoding PEM block from private key")
+	}
+
+	priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("file-decrypt: error parsing private key: %v", err)
+	}
+
+	return priv, nil
+}
+
+// Decrypts a file encrypted with AES (key length depends on input) in CBC block
+// chaining mode and PKCS#7 padding. The provided key must be encoded in base16,
+// and the first block of the input is the IV. The output is a pipe reader that
+// can be used to stream the decrypted file.
+func DecryptAESCBC(filePath string, privateKey *rsa.PrivateKey, encryptedKeyFile string) (io.ReadCloser, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file: %w", err)
+	}
+
+	reader := ioutil.NopCloser(bufio.NewReader(file))
+
+	iv := make([]byte, aes.BlockSize)
+	if _, err := io.ReadFull(reader, iv); err != nil {
+		return nil, fmt.Errorf("error reading iv from input: %w", err)
+	}
+
+	return DecryptAESCBCWithIV(reader, privateKey, encryptedKeyFile, iv)
+}
+
+func DecryptAESCBCWithIV(reader io.ReadCloser, privateKey *rsa.PrivateKey, encryptedKeyB64 string, iv []byte) (io.ReadCloser, error) {
+
+	encryptedKey, err := base64.StdEncoding.DecodeString(encryptedKeyB64)
+	if err != nil {
+		glog.Fatalf("Error decoding base64 encoded key: %v", err)
```

Comment:
Also needs to be a returned error. Fatal will crash the process. Might be worth going round the PR diff to check if there aren't any other fatal logs like these